src/java.base/share/classes/sun/security/util/Debug.java

```diff
@@ -153,12 +175,51 @@ public static Debug getInstance(String option, String prefix)
         if (isOn(option)) {
             Debug d = new Debug();
             d.prefix = prefix;
+            d.printThreadDetails = getConfigInfo(option, "+thread");
+            d.printDateTime = getConfigInfo(option, "+timestamp");
+            if (d.printDateTime && !dateTimeFormatInitialized) {
+                // trigger loading of Locale service impl now to avoid
```

Comment:
You may want to try the test case added in JDK-8280890 with debugging enabled to make sure you don't get a similar recursion issue.

Reply:
Interesting. It seems any attempt to calls like `Date.toString()` early on that triggers a `TimeZone` lookup is going to potentially cause this. A possible workaround is to change `CertificateValidity.toString()` to emit dates as `Date.toInstant().toString()` which is similar to what I did in the fix for JDK-8280890. But this is a separate issue.